src/ast/InheritanceForest.java

```diff
@@ -116,12 +116,13 @@ public InheritanceForest(Program prog) {
 	}
 	
 	private boolean legalClassDecl(ClassDecl cls, String superName, Set<String> allNames){
-		//checks:
+		// checks:
 		//1. super is defined
 		//2. super is not self
-		//3. super is not main
+		//3. super is not main (req 2)
 		//4. class name is not a repetition
-		//5. super is defined before self
+		//5. super is defined before self (req 1)
+		// TODO: check for cycles in forest (req 1)
```

Comment:
no need to explicitly check for cycles, the checking of super defined before child takes care of that